src/validation.ts

```diff
@@ -144,14 +144,35 @@ export const validateTLSServerName = (serverName: string) => {
 };
 
 export const validateTLSCACert = (caCert: string) => {
+  if (!caCert) {
+    return undefined;
+  }
+  if (caCert.indexOf('-----BEGIN CERTIFICATE-----') < 0 || caCert.indexOf('-----END CERTIFICATE-----') < 0) {
+    return 'Certificate must be in the PEM format.';
+  }
   return undefined;
 };
 
 export const validateTLSClientCert = (clientCert: string) => {
+  if (!clientCert) {
+    return undefined;
+  }
+  if (clientCert.indexOf('-----BEGIN CERTIFICATE-----') < 0 || clientCert.indexOf('-----END CERTIFICATE-----') < 0) {
+    return 'Certificate must be in the PEM format.';
+  }
   return undefined;
 };
 
 export const validateTLSClientKey = (clientKey: string) => {
+  if (!clientKey) {
+    return undefined;
+  }
+  if (
+    clientKey.indexOf('-----BEGIN RSA PRIVATE KEY-----') < 0 ||
```

Comment:
this is probably too restrictive as there are keys other than RSA

The PEM start line starts with "-----BEGIN " and the end line starts with "-----END ". Both are followed by a type and end with "-----". Maybe check that the type ends with "PRIVATE KEY".